geth/transactions/queue/queue.go

```diff
@@ -189,105 +160,77 @@ func (q *TxQueue) Get(id common.QueuedTxID) (*common.QueuedTx, error) {
 	if tx, ok := q.transactions[id]; ok {
 		return tx, nil
 	}
+	return nil, ErrQueuedTxIDNotFound
+}
+
+// LockInprogress returns transcation and locks it as inprogress
+func (q *TxQueue) LockInprogress(id common.QueuedTxID) (*common.QueuedTx, error) {
```

Comment:
Do we need to return `common.QueuedTx`?

Reply:
i wanted to avoid doing get and then lock, particularly i was thinking about SELECT FOR UPDATE sql statement when i added this method

maybe naming is unfortunate, though

Reply:
I can't find where it is used except for a test file. 

If we have `LockInProgress()` and `Get()` and call it in this order everything should be ok without race conditions, right?

Reply:
> maybe naming is unfortunate, though
Agree. I had to go to the function definition to understand what does it mean. Maybe `GetAndLock`?

@adambabik it's used in `(*Manager).CompleteTransaction()`:
```go
// CompleteTransaction instructs backend to complete sending of a given transaction.
// TODO(adam): investigate a possible bug that calling this method multiple times with the same Transaction ID
// results in sending multiple transactions.
func (m *Manager) CompleteTransaction(id common.QueuedTxID, password string) (hash gethcommon.Hash, err error) {
    log.Info("complete transaction", "id", id)
    tx, err := m.txQueue.LockInprogress(id)
    if err != nil {
        log.Warn("error getting a queued transaction", "err", err)
        return hash, err
    }
```
BTW, is it solving bug mentioned in TODO?

Reply:
Oh yeah, this comment is not relevant anymore. The problem was that we were not tracking which transaction is in progress. The fix was introduced before this refactoring and we are still good so let's remove it.

I would just go with `LockInProgress() error` and use `Get()`. This code is not a bottleneck so readability vs preoptimization.